### PR TITLE
Common: method deprecations

### PR DESCRIPTION
**Pull request by ScottyPoi**
1. VM / Client / Devp2p / Block / Common: Rename hardforkBlockBN() to hardforkBlock()
2. Common: Remove deprecated hardforkBlock()
3. Common: Update build:docs
4. Rename hardforkBlock in tests 
5. Fix tests by using .eqn to test BN's against numbers.

### CODE REVIEW

**Comment by ScottyPoi** [comment]
I'm not sure why the 'hardhat' test fails

**Comment by acolytec3** [comment]
It's not your code. Develop is behind master and the package numbers are out of date.  We need to rebase develop and they should solve this problem.

File: packages/common/tests/hardforks.spec.ts [hunk 1]
```diff
@@ -225,14 +224,13 @@ tape('[Common]: Hardfork logic', function (t: tape.Test) {
...
-    st.equal(c.hardforkBlock(Hardfork.Berlin), 12244000, msg)
-    st.ok(c.hardforkBlockBN(Hardfork.Berlin)!.eq(new BN(12244000)), msg)
+    st.ok(c.hardforkBlock(Hardfork.Berlin)!.eqn(12244000), msg)
 
     msg = 'should return null for unscheduled hardfork'
     // developer note: when Shanghai is set,
     // update this test to next unscheduled hardfork.
     st.equal(c.hardforkBlock(Hardfork.Shanghai), null, msg)
-    st.equal(c.hardforkBlockBN(Hardfork.Shanghai), null, msg)
+    st.equal(c.hardforkBlock(Hardfork.Shanghai), null, msg)
```
Comment (acolytec3) [inline packages/common/tests/hardforks.spec.ts:233]: I think you can safely remove this line as it duplicates the one directly above it.

**Review by acolytec3 — COMMENTED** [review]
Looking good so far!  I left one comment to clean up a duplicate line.

Also, once we rebase develop on current master, the hardhat test should start passing again unless they use the `hardforkBlockBN` method in their code.

@holgerd77 In the next breaking release, should we entirely remove `hardforkBlockBN` or just deprecate it and internally rewrite it to use the now rewritten `hardforkBlock`?

**Comment by holgerd77** [comment]
@acolytec3 The old `hardforkBlockBN` names will be removed entirely.

(also, side note: it will likely be the case that this methods will also be transitioned from the `BN.js` to `BigInt` usage. This step is just for separation of concerns.)

I will close this PR here, have mailed Scotty on this, still some things to address.

**Pull request closed without merging**